### PR TITLE
Fixes #1736: Improved quality of TypeError exceptions raised for WBEMConnection ops args

### LINKED ISSUE #1736: Improve exception message when passing int objects to method params
Issue #1733 revealed that the TypeError that is raised when passing int objects is not easy to understand and needs to be improved:

```
Params = [
    ('ElementType', 2),
    ('InPool', sp_paths[0]),
    ('Size', 1024000),
    ('TheElement', None)
]
result, outparams = conn.InvokeMethod(
    'CreateOrModifyElementFromStoragePool',
    os_inst_path,
    Params)
```

This raises the following `TypeError`, due to the fact that the 'ElementType' parameter is specified as a Python `int` object in the input parameters of the method, when it should have been a `pywbem.Uint32` object:

```
Traceback (most recent call last):
File "", line 1, in
File "pywbem/cim_operations.py", line 4081, in InvokeMethod
result = self._methodcall(MethodName, ObjectName, Params, **params)
File "pywbem/cim_operations.py", line 1995, in _methodcall
ptuple = (p[0], p[1], infer_type(p[1]),
File "pywbem/cim_operations.py", line 1948, in infer_type
_format("Unsupported parameter type {0}", type(obj)))
TypeError: Unsupported parameter type <type 'int'>
```

The exception message does not really allow a user to understand what it even complains about, that they can correct it themselves, or how that would be done.

This should be improved.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

**Comment by andy-maier** [comment]
Karl, your comment seems to have gotten lost...

File: pywbem/cim_operations.py [hunk 1]
```diff
@@ -1941,11 +1944,16 @@ def infer_type(obj):
...
+                return infer_type(obj[0], param_name) if obj else None
             elif obj is None:
                 return None
+            if isinstance(obj, int):
+                hint = " (use a CIM integer type such as pywbem.Uint32)"
+            else:
+                hint = ""
             raise TypeError(
-                _format("Unsupported parameter type {0}", type(obj)))
+                _format("Method parameter {0!A} has unsupported type {1}{2}",
```
Comment (KSchopmeyer) [inline pywbem/cim_operations.py]: Suggest rewording this to something like "Method Parameter {0!A} type {1} cannot be used to infer a valid CIM data type{2}.

Comment (andy-maier) [reply pywbem/cim_operations.py]: DONE - changed text to:
"Method parameter {0!A} has type {1} which cannot be used to infer a valid CIM data type{2}"

**Comment by andy-maier** [comment]
Resolved the comment.
